### PR TITLE
feat(renderer): set `useTypeImports` to render all imports as type imports

### PR DESCRIPTION
**Pull request by BradenM**
Modified the import rendering to utilize explicit type imports.

Aside from the build performance/related benefits in using explicit type imports, it is often a requirement with certain tsconfig setups (such as using `verbatimModuleSyntax` in ts 5.0).

I am making these changes under the assumption that it is indeed true to say that `jsii-struct-builder` would only ever require type imports. If I am missing something and this is not true, please do let me know :smile:

### CODE REVIEW

**Review by mrgrain — COMMENTED** [review]
I tend to agree, but I'm not familiar with that feature at all.
Can you make it an optional setting that defaults to using type imports.
>Aside from the build performance/related benefits in using explicit type imports

**Comment by BradenM** [comment]
@mrgrain 
This short post gives a pretty good overview:
https://typescript-eslint.io/blog/consistent-type-imports-and-exports-why-and-how

My understanding as to where the build performance comes in is the fact that it allows typescript to "blindly" drop `type` imports instead of having to figure out whether an import is required during runtime or not. Although this also depends on a user's typescript config.

> What I could find was that it's mostly useful to work around some edge case behavior.

I believe what you may be referring to is using `type` imports to circumvent unexpected side effects when importing a module. I believe this was the core behind the original push for it, as before it was tricky if not impossible to try to import a module (even if purely for type purposes) if it had unwanted side effects. Personally, I would argue that this was a bit more than just an edge case, but that may just be the pain of dealing with it in the past speaking :smile:

> Can you make it an optional setting that defaults to using type imports (but also see below).
All this said, I can definitely look into implementing this if it is what you prefer.

As for jsii, I don't see why it wouldn't work. But until recently, I have primary just been a consumer of jsii libraries and am definitely not familiar enough yet to give a confident answer to that one :smile:

**Comment by mrgrain** [comment]
> This short post gives a pretty good overview:
> https://typescript-eslint.io/blog/consistent-type-imports-and-exports-why-and-how

Thanks! Appreciate the link.



> As for jsii, I don't see why it wouldn't work. But until recently, I have primary just been a consumer of jsii libraries and am definitely not familiar enough yet with the inner workings to give a confident answer to that one 😄

Yep, and I don't expect you to. I was thinking out aloud really. Let's go with my original suggestion of a new option and defaulting it to types. I can do a manual jsii test run over the weekend and/or setup an integ test of sorts.

**Comment by mrgrain** [comment]
This looks good. I haven't tested it yet with jsii and am keen to make it the default if possible. Please ping me if I don't get to it within a week 🙈

**Comment by BradenM** [comment]
@mrgrain No rush, just had forgotten to ping you as requested... so here is a *slightly* late one :smile:

**Review by mrgrain — APPROVED** [review]

**Comment by mrgrain** [comment]
> @mrgrain No rush, just had forgotten to ping you as requested... so here is a _slightly_ late one 😄

Thanks! I mean I could have remembered... 😅 

I'll merge it like this to do some tests. If it all works as expected we can do a BC to change the default.

File: test/projen/ts-interface.test.ts [hunk 1]
```diff
@@ -94,6 +94,26 @@ test('can override package imports', () => {
   );
 });
 
+test('can use explicit type imports', () => {
+  // Arrange
+  const project = new TestProject();
+  const spec = findInterface(
+    'projen.typescript.TypeScriptProjectOptions',
+    false
```
Comment (mrgrain) [inline test/projen/ts-interface.test.ts:102]:
```suggestion
    false,
```

File: test/projen/ts-interface.test.ts [hunk 2]
```diff
@@ -94,6 +94,26 @@ test('can override package imports', () => {
...
+
+  // ACT
+  new TypeScriptInterfaceFile(project, 'interface.ts', spec, {
+    useTypeImports: true,
+  });
+  const renderedFile = synthSnapshot(project)['interface.ts'];
+
+  // ASSERT
+  expect(renderedFile).toContain(
+    "import type { javascript, typescript } from '../';"
```
Comment (mrgrain) [inline test/projen/ts-interface.test.ts:113]:
```suggestion
    "import type { javascript, typescript } from '../';",
```

**Pull request merged**